Sources/SwiftyMarkdown/SwiftyMarkdown.swift

```diff
@@ -27,12 +27,45 @@ public enum CharacterStyle : CharacterStyling {
 	case referencedLink
 	case referencedImage
 	case strikethrough
+    #if os(macOS)
+    case color(color: NSColor)
+    case font(font: NSFont)
+    #else
+    case color(color: UIColor)
+    case font(font: UIFont)
+    #endif
 	
 	public func isEqualTo(_ other: CharacterStyling) -> Bool {
 		guard let other = other as? CharacterStyle else {
 			return false
 		}
-		return other == self 
+        
+        switch (self, other) {
+        case (.color, .color):
+            return true
+        case (.font, .font):
+            return true
+        case (.none, .none):
+            return true
+        case (.bold, .bold):
+            return true
+        case (.italic, .italic):
+            return true
+        case (.code, .code):
+            return true
+        case (.link, .link):
+            return true
+        case (.image, .image):
+            return true
+        case (.referencedLink, .referencedLink):
+            return true
+        case (.referencedImage, .referencedImage):
+            return true
+        case (.strikethrough, .strikethrough):
+            return true
```

Comment:
I find Swift's comma-separated notation really nice to this use case:

```suggestion
        case (.color, .color),
            (.font, .font),
            (.none, .none),
            (.bold, .bold),
            (.italic, .italic),
            (.code, .code),
            (.link, .link),
            (.image, .image),
            (.referencedLink, .referencedLink),
            (.referencedImage, .referencedImage),
            (.strikethrough, .strikethrough):
            return true
```

Reply:
it triggers my inner monk a little bit since it loses the symmetry of the the other format, but I don't have a strong opinion on that. we can change it 😅

Reply:
It's also fine to leave it like this :+1: